### PR TITLE
Search Wiki snippet for PowerPoint

### PR DESCRIPTION
**Pull request by cakriwut**
**#Description**  
Demonstrate searching Wikipedia API based on selected text in the presentation.

**#Target Host**
PowerPoint

### CODE REVIEW

File: samples/powerpoint/custom-functions/search-wikipedia-api.yaml [hunk 1]
```diff
@@ -0,0 +1,221 @@
+order: 1
+id: powerpoint-custom-functions-search-wikipedia-api
```
Comment (AlexJerabek) [inline samples/powerpoint/custom-functions/search-wikipedia-api.yaml]: Custom functions has a very specific meaning in Excel, so we should avoid using that term in this sample (lest people think PowerPoint supports [custom functions](https://docs.microsoft.com//office/dev/add-ins/excel/custom-functions-overview)).

I'd recommend changing the category and folder to "Scenarios". This sample shows the common scenario of calling an external web API for data.

File: samples/powerpoint/custom-functions/search-wikipedia-api.yaml [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+name: Search Wikipedia
+description: Search Wikipedia based on the selected text in the presentation.
+author: cakriwut
+host: POWERPOINT
+api_set: {}
+script:
+    content: |
+        declare var moment: any;
```
Comment (AlexJerabek) [inline samples/powerpoint/custom-functions/search-wikipedia-api.yaml]: Unless it's explicitly a JavaScript sample, snippets in this repo try to conform to TypeScript standards. Let's change `var` to `let` or `const` where applicable.

File: samples/powerpoint/custom-functions/search-wikipedia-api.yaml [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+        $("#search").click(run);
+
+        async function run() {
+          Office.context.document.getSelectedDataAsync(Office.CoercionType.Text, {}, getSelectedText);
+        }
+
+        function getSelectedText(result) {
```
Comment (AlexJerabek) [inline samples/powerpoint/custom-functions/search-wikipedia-api.yaml]: In order to help others better learn from your snippet, could you add a brief description before each function? A short sentence or two will help people understand what your code is doing and parse it faster.

File: samples/powerpoint/custom-functions/search-wikipedia-api.yaml [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+          $("#result").empty();
+          $("#result").append('<ul class="ms-List"></ul>');
+          if (result.status === "succeeded") {
+            searchWiki(result.value);
+          } else {
+            console.log("failed");
```
Comment (AlexJerabek) [inline samples/powerpoint/custom-functions/search-wikipedia-api.yaml]: Please expand the error statement to explain what part of the call failed, as best as you're able.

File: samples/powerpoint/custom-functions/search-wikipedia-api.yaml [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+        }
+    language: typescript
+template:
+    content: |-
+        <header class="ms-welcome__header ms-bgColor-neutralLighter" style="text-align: center">
+          <h1 class="ms-font-su">Search Wikipedia</h1>
+        </header>
+
+        <section class="ms-font-m">
+          <p>This sample shows how to query external API using selected text in the presentation.</p>
```
Comment (AlexJerabek) [inline samples/powerpoint/custom-functions/search-wikipedia-api.yaml]: A couple grammar suggestions. Consider changing this to:
"This sample shows how to query an external API (Wikipedia) with the text currently selected in the presentation."

That helps highlight the interactions between the add-in and the presentation, which is the exciting part.

File: samples/powerpoint/custom-functions/search-wikipedia-api.yaml [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+          <p>The result then will be displayed in the task pane.</p>
```
Comment (AlexJerabek) [inline samples/powerpoint/custom-functions/search-wikipedia-api.yaml]: I don't think you need this line about where the information is displayed.

File: samples/powerpoint/custom-functions/search-wikipedia-api.yaml [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+        </section>
+
+        <section class="setup ms-font-m">
+          <h3>Try it out</h3>
+          <p>The search result will be provided by Wikipedia API. Select on any word/sentence in the presentation and, click "Search"</p>
```
Comment (AlexJerabek) [inline samples/powerpoint/custom-functions/search-wikipedia-api.yaml]: Consider changing this line to
"Select any text in the presentation and press <b>Search</b> to see related Wikipedia entries."

File: samples/powerpoint/custom-functions/search-wikipedia-api.yaml [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+          <ol>
+            <li>In the presentation, select any word/sentence</li>
+            <li>Click Search button: <br /><br />
```
Comment (AlexJerabek) [inline samples/powerpoint/custom-functions/search-wikipedia-api.yaml]: With the sentence suggested above, you can get rid of these instructions.

File: samples/powerpoint/custom-functions/search-wikipedia-api.yaml [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+              <button id="search" class="ms-Button">
+                <span class="ms-Button-label">Run</span>
```
Comment (AlexJerabek) [inline samples/powerpoint/custom-functions/search-wikipedia-api.yaml]: Please change this button name to "Search" to match the description.

File: samples/powerpoint/custom-functions/search-wikipedia-api.yaml [hunk 10]
```diff
@@ -0,0 +1,221 @@
+order: 1
+id: powerpoint-custom-functions-search-wikipedia-api
+name: Search Wikipedia
+description: Search Wikipedia based on the selected text in the presentation.
```
[the remark this answers is not in the document]
Comment (cakriwut) [reply samples/powerpoint/custom-functions/search-wikipedia-api.yaml]: Updated.

File: samples/powerpoint/custom-functions/search-wikipedia-api.yaml (hunk 1 above)
Comment (cakriwut) [reply samples/powerpoint/custom-functions/search-wikipedia-api.yaml]: Move and update to scenarios folder

File: samples/powerpoint/custom-functions/search-wikipedia-api.yaml (hunk 2 above)
Comment (cakriwut) [reply samples/powerpoint/custom-functions/search-wikipedia-api.yaml]: Updated

File: samples/powerpoint/custom-functions/search-wikipedia-api.yaml (hunk 3 above)
Comment (cakriwut) [reply samples/powerpoint/custom-functions/search-wikipedia-api.yaml]: Updated.

File: samples/powerpoint/custom-functions/search-wikipedia-api.yaml (hunk 4 above)
Comment (cakriwut) [reply samples/powerpoint/custom-functions/search-wikipedia-api.yaml]: Updated.

File: samples/powerpoint/custom-functions/search-wikipedia-api.yaml (hunk 5 above)
Comment (cakriwut) [reply samples/powerpoint/custom-functions/search-wikipedia-api.yaml]: Updated.

File: samples/powerpoint/custom-functions/search-wikipedia-api.yaml (hunk 6 above)
Comment (cakriwut) [reply samples/powerpoint/custom-functions/search-wikipedia-api.yaml]: Updated.

File: samples/powerpoint/scenarios/searches-wikipedia-api.yaml [hunk 11]
```diff
@@ -0,0 +1,233 @@
+order: 1
+id: powerpoint-scenarios-searches-wikipedia-api
+name: Searches Wikipedia
+description: Searches Wikipedia based on the selected text in the presentation.
+author: cakriwut
+host: POWERPOINT
+api_set: {}
+script:
+    content: |
+        declare let moment: any;
```
Comment (davidchesnut) [inline samples/powerpoint/scenarios/searches-wikipedia-api.yaml]: @cakriwut thanks for the updates! Would you mind also adding your copyright here at the top of the sample? We like to give credit for every contribution! The typical format would be:
/*
 * Copyright (c) Riwut Libinuko. All rights reserved. Licensed under the MIT license.
 */
or similar...
Thanks!

**Comment by AlexJerabek** [comment]
@cakriwut, Thanks for the updates! We just need the necessary copyright information, as @davidchesnut suggested. I am happy to add that and submit this PR on your behalf, if you'd like.

File: samples/powerpoint/scenarios/searches-wikipedia-api.yaml (hunk 11 above)
Comment (cakriwut) [reply samples/powerpoint/scenarios/searches-wikipedia-api.yaml]: Thanks for the acceptance.

**Comment by cakriwut** [comment]
> @cakriwut, Thanks for the updates! We just need the necessary copyright information, as @davidchesnut suggested. I am happy to add that and submit this PR on your behalf, if you'd like.

@AlexJerabek  - thanks. I have updated.

**Pull request merged**